client/post-editor/editor-term-selector/index.jsx

```diff
@@ -26,21 +30,23 @@ class EditorTermSelector extends Component {
 	}
 
 	onTermsChange( selectedTerm ) {
-		const { postTerms, taxonomyName } = this.props;
+		const { postTerms, taxonomyName, siteId, postId } = this.props;
 		const terms = cloneDeep( postTerms ) || {};
-		const taxonomyTerms = terms[ taxonomyName ] || {};
 
-		if ( taxonomyTerms[ selectedTerm.name ] ) {
-			delete taxonomyTerms[ selectedTerm.name ];
+		// map call transforms object returned by API into an array
+		const taxonomyTerms = map( terms[ taxonomyName ] ) || [];
```

Comment:
Minor: Looks like both `_.map` and `_.toArray` are smart enough to result in an empty array when passed an undefined argument:

```
wp-calypso (add/term-tree-selector/add-term) $ node
> var map = require( 'lodash/map' );
undefined
> map( undefined );
[]
> var toArray = require( 'lodash/toArray' );
undefined
> toArray( undefined );
[]
```